modules/container_runtime/main.tf

```diff
@@ -65,5 +68,172 @@ resource "aws_ssm_parameter" "ssm_container_runtime_kms_key_id" {
+# ---------------------------------------------------------------------------------------------------------------------
+# tfsec:ignore:aws-ecs-enable-container-insight
+resource "aws_ecs_cluster" "ec2_cluster" {
+  count = local.create_ecs_with_ec2 ? 1 : 0
+  name  = var.container_runtime_name
+
+  configuration {
+    execute_command_configuration {
+      logging = "NONE"
+    }
+  }
+}
+
+locals {
+  user_data = <<-EOT
+    #!/bin/bash
+    cat <<'EOF' >> /etc/ecs/ecs.config
+    ECS_CLUSTER=${var.container_runtime_name}
+    ECS_LOGLEVEL=debug
+    EOF
+  EOT
+}
+
+resource "aws_ecs_cluster_capacity_providers" "ecs_ec2_cap_provider" {
+  count              = local.create_ecs_with_ec2 ? 1 : 0
+  cluster_name       = aws_ecs_cluster.ec2_cluster[0].name
+  capacity_providers = [aws_ecs_capacity_provider.ecs_ec2_capacity_provider[0].name]
+
+  default_capacity_provider_strategy {
+    base              = 20
+    weight            = 60
+    capacity_provider = aws_ecs_capacity_provider.ecs_ec2_capacity_provider[0].name
+  }
+}
+
+resource "aws_ecs_capacity_provider" "ecs_ec2_capacity_provider" {
+  count = local.create_ecs_with_ec2 ? 1 : 0
+  name  = "ec2_capacity_provider"
+
+  auto_scaling_group_provider {
+    auto_scaling_group_arn         = aws_autoscaling_group.ecs_ec2_asg[count.index].arn
+    managed_termination_protection = "ENABLED"
+
+    managed_scaling {
+      maximum_scaling_step_size = 1
+      minimum_scaling_step_size = 1
+      status                    = "ENABLED"
+      target_capacity           = 60
+    }
+  }
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# Auto scaling group and Launch template
+# ---------------------------------------------------------------------------------------------------------------------
+resource "aws_autoscaling_group" "ecs_ec2_asg" {
+  count               = local.create_ecs_with_ec2 ? 1 : 0
+  name                = "ecs_ec2_autoscaling"
+  vpc_zone_identifier = data.aws_subnets.private_subnets.ids #aws_subnet.private_subnets.*.id
+
+  launch_template {
+    id      = aws_launch_template.ecs_ec2_launch_template[count.index].id
+    version = aws_launch_template.ecs_ec2_launch_template[count.index].latest_version
+  }
+
+  protect_from_scale_in     = true
+  desired_capacity          = 2
+  min_size                  = 0
+  max_size                  = 2
+  health_check_grace_period = 300
+  health_check_type         = "ELB"
+  wait_for_capacity_timeout = "10m"
+  wait_for_elb_capacity     = 2
+
+
+  lifecycle {
+    create_before_destroy = true
+    ignore_changes        = [load_balancers, target_group_arns]
+  }
+  tag {
+    key                 = "AmazonECSManaged"
+    value               = true
+    propagate_at_launch = true
+  }
+}
+resource "aws_launch_template" "ecs_ec2_launch_template" {
+  count                  = local.create_ecs_with_ec2 ? 1 : 0
+  name_prefix            = "ecs_ec2_container_instance"
```

Comment:
rename into terra3_ecs_ec2_container_instance to more easily identify terra3 resources.

Reply:
done